OpenRVS/classes/OpenServerList.uc

```diff
@@ -47,68 +50,62 @@ function Init(OpenMultiPlayerWidget Widget, optional string W, optional string F
 	httpc.SendRequest("http://" $ WebAddress $ "/" $ FileName);
 }
 
-//when connection is finished, check the length of each received line.
-//lines too large will crash/cause other issues
-//so split lines into manageable size
-//then parse them to find server info and send back to the multiplayer menu.
-//strips out extra info on the ends of the string and returns a server string in this format:
-//a server name",IP="44.434.4.434:7777",Locked=false,GameMode="Adver
-//the function in OpenMultiPlayerWidget will parse this further to get the needed info.
-//1.3 - NO LONGER parsed in OpenMultiPlayerWidget, parsed below
-function ParseServersINI(OpenHTTPClient.HttpResponse resp)
+// Receives and detects type for an HTTP response from the server list provider.
+// It is also responsible for sending servers to OpenMultiPlayerWidget.
+function ParseServers(OpenHTTPClient.HttpResponse resp)
 {
 	local array<string> lines;
-	local string line;
-	local int i;
 
 	httpc = none;//done with HTTP client
 
 	//Check for errors.
 	if (resp.Code != 200)
 	{
 		class'OpenLogger'.static.Error("failed to retrieve servers over http", self);
-		NoServerList();
+		SendServersUpstream(GetFallbackServers());
+		return;
 	}
 
-	// Convert each line to CSV K=V format (i.e. "a=b,c=d,e=f")
+	// Determine response type.
 	class'OpenString'.static.Split(resp.Body, Chr(10), lines);
-	lines.Remove(0, 1);//skips header line (specific to INI parser)
-	for (i=0; i<lines.Length; i++)
+	if (Left(lines[0], Len(INI_HEADER_LINE)) == INI_HEADER_LINE)
 	{
-		line = lines[i];
-		line = Mid(line, 12);//trim up to "ServerName="
-		line = Left(line, Len(line)-1);//remove )
-		lines[i] = line;//overwrite
+		SendServersUpstream(ParseServersINI(lines));
+		return;
 	}
+	// Add new supported types (e.g. CSV) here.
 
-	if ( lines.length > 0 )
-	{
-		class'OpenLogger'.static.Info("loading server list from URL", self);
-		ServerListSuccess(lines);//1.3 parsing moved here
-	}
-	else
-		NoServerList();//1.3 noserverlist moved here
+	// Fall back to local file.
+	class'OpenLogger'.static.Error("unknown server list response type", self);
+	SendServersUpstream(GetFallbackServers());
 }
 
-//1.3
-//function moved here from openmultiplayerwidget
-//in the widget, the config list of servers was getting saved and loaded in the user.ini
-//and filling it with needless garbage?
-//moving it here should prevent it from saving automatically to user.ini
-//and prevent it from trying to load the glitchy lines?
-//function parses an array of strings to get server info, then will send each server item to widget
-function ServerListSuccess(array<string> List)
+// Parses a server list in the INI file format.
+function array<AServer> ParseServersINI(array<string> lines)
 {
+	local string line;//server list line "ServerName=(a=b,c=d,e=f)"

```

Comment:
shouldn't this comment be:
//server list line "ServerList=(a=b,c=d,e=f)"
because ServerName is one of the internal properties of the ServerList array?